applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.cpp

```diff
@@ -284,8 +404,15 @@ void DVMSDEMCoupled<TElementData>::AddMassStabilization(
     AGradN *= density;
 
     const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    double viscosity = this->GetAtCoordinate(rData.DynamicViscosity, rData.N);
+    Matrix permeability = this->GetAtCoordinate(rData.Permeability, rData.N);
+    Matrix inv_permeability = ZeroMatrix(Dim, Dim);
 
-    double W = rData.Weight * tau_one * density; // This density is for the dynamic term in the residual (rho*Du/Dt)
+    double det_permeability = MathUtils<double>::Det(permeability);
+    MathUtils<double>::InvertMatrix(permeability, inv_permeability, det_permeability, -1.0);
+
+    double W = rData.Weight * density; // This density is for the dynamic term in the residual (rho*Du/Dt)
+    Matrix sigma = inv_permeability * viscosity;
```

Comment:
Can't you reuse `inv_permeability` to avoid allocating another matrix?

Reply:
Yes I could use it, but I prefer to define sigma for the sake of clarity.

Reply:
If this is the case, I'd declare `sigma` above and then use it as input for the `InvertMatrix`. Such dynamic allocations of `Matrix` are way inefficient (OFC I'm not blocking because of this, so feel free to ignore this comment).

Reply:
Ok, I have modified the inv_permeability variable